### PR TITLE
Add chmod 755 for envoy in proxy docker rule.

### PR DESCRIPTION
**Pull request by diemtvu**
https://github.com/istio/istio/issues/4217

### CODE REVIEW

**Comment by ldemailly** [comment]
> Please add the chmod in the makefile, not in docker.

why ? (and/or why not both)

File: pilot/docker/Dockerfile.proxy [hunk 1]
```diff
@@ -11,5 +11,8 @@ COPY envoy_mixer.json      /etc/istio/proxy/envoy_mixer.json
 COPY envoy_mixer_auth.json /etc/istio/proxy/envoy_mixer_auth.json
 COPY envoy_bootstrap_tmpl.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
 
+# Allow non-root user can run envoy.
+RUN chmod 755 /usr/local/bin/envoy
```
Comment (ldemailly) [inline pilot/docker/Dockerfile.proxy]: I would move that just below the ADD if we are picky

File: pilot/docker/Dockerfile.proxy [hunk 2]
```diff
@@ -11,5 +11,8 @@ COPY envoy_mixer.json      /etc/istio/proxy/envoy_mixer.json
 COPY envoy_mixer_auth.json /etc/istio/proxy/envoy_mixer_auth.json
 COPY envoy_bootstrap_tmpl.json /var/lib/istio/envoy/envoy_bootstrap_tmpl.json
 
+# Allow non-root user can run envoy.
```
Comment (ldemailly) [inline pilot/docker/Dockerfile.proxy]: replace "can" by "to"

**Comment by costinm** [comment]
The output of the make should have the right permissions - make shouldn't
create files
with bad permission.

If make creates with right permission - it doesn't have to be duplicated in
the dockerfiles.
It may not hurt - but the root problem is what 'tar xf' creates, and we
should fix it at the root.
Otherwise we'll also need to do chmod in any other thing that uses the
binary.

**Comment by ldemailly** [comment]
this is a short term fix, the issue #4217 covers better fixes options
but given the makefiles are in a constant state of change and various brokenness I think this fix is good for the immediate problem and I don't see any downside to it

**Comment by ldemailly** [comment]
ahem, you forced pushed over my fixes :-)

**Comment by diemtvu** [comment]
whoops. sorry.

**Comment by costinm** [comment]
Well, I still believe this is the wrong fix - and would appreciate a follow
up to fix the actual permission.

If you really want to change the Dockerfile -
 ADD --chown ...
would also work.

**Comment by ldemailly** [comment]
>  ADD --chown ...

isn't going to help with permissions/modes; we shouldn't have the istio-user own the binary, that opens up the possibility of corrupting the executable etc

**Comment by ldemailly** [comment]
merging this even if it's imperfect (should be /* not just envoy etc...)

**Pull request merged**